src/python/pants/engine/internals/build_files.py

```diff
@@ -103,29 +99,29 @@ async def parse_address_family(
     return AddressFamily.create(directory.path, address_maps)
 
 
-def _raise_did_you_mean(address_family: AddressFamily, name: str, source=None) -> None:
-    names = [a.address.target_name for a in address_family.addressables]
+def _did_you_mean_exception(address_family: AddressFamily, name: str) -> ResolveError:
+    names = [
+        addr.target_name or os.path.basename(addr.spec_path)
```

Comment:
This was a bug from the previous model change for Address. It was possible for us to output `None` as an option because `target_name` is Optional.

Reply:
`target_name` is a property that does this same calculation, so I don't think so?

But another possibility here would be for this to say something like:
```
There was no default target (ie, no target with the default `name=` '{name}') in the directory '{address_family.namespace}'...
```

... just to continue to push the concept of default targets.

Reply:
It might help. If you don't know that passing no `name` will get you a default `name`, you might see this error, and not know what the name of the target you are looking for. But yea, fine.